### PR TITLE
Damage fishing rod when catching the hook on a block, then reeling it in

### PR DESCRIPTION
**Pull request by Bond-009**
#4132

### CODE REVIEW

**Comment by Bond-009** [comment]
I did, and it worked in all scenarios that I tested it in.

File: src/Items/ItemFishingRod.h [hunk 1]
```diff
@@ -234,6 +234,10 @@ class cItemFishingRodHandler :
 				a_Player->UseEquippedItem(1);
 				cRoot::Get()->GetPluginManager()->CallHookPlayerFished(*a_Player, Drops);
 			}
+			else if (!IsBlockWater(a_World->GetBlock(FloaterInfo.GetPos())))
```
Comment (peterbell10) [inline src/Items/ItemFishingRod.h]: This doesn't seem right.  If the floater is in the air that's still not water but it shouldn't be taking damage.  I think `IsSolid` would be better, or whatever the floater's physics uses for collisions would probably be best.

Comment (Bond-009) [reply src/Items/ItemFishingRod.h]: Thanks, I didn't think about that.

**Comment by peterbell10** [comment]
I tried it in game and it never damages the fishing rod.  I'm guessing @bearbin's right and the floater does indeed sit outside the block.

**Comment by bearbin** [comment]
So the code should instead be checking for an air block, and the code for floater placement double-checked to confirm.

**Comment by Bond-009** [comment]
Did you test it before before the changed to address your comment or after (or both)?

**Comment by peterbell10** [comment]
@bearbin I don't think there's any way around checking a (small) volume like in #4136.

@Bond-009 I tested both: at first it was always damaged, now it is never damaged.

**Comment by Bond-009** [comment]
@peterbell10 is this a good solution?

**Comment by bearbin** [comment]
I don't believe this will handle hooks caught on vertical walls - will it?

**Pull request merged**